kubernetes/cache/cache.go

```diff
@@ -202,5 +204,78 @@ func (in *kialiCacheImpl) IsAmbientEnabled(cluster string) bool {
 	return check
 }
 
+type namespacesKey struct {
+	cluster string
+	token   string
+}
+
+func (c *kialiCacheImpl) GetNamespace(cluster string, token string, namespace string) (models.Namespace, bool) {
+	c.namespacesLock.RLock()
+	defer c.namespacesLock.RUnlock()
+
+	key := namespacesKey{cluster: cluster, token: token}
+	namespaces, found := c.namespaceStore.Get(key)
+	if !found {
+		return models.Namespace{}, false
+	}
+
+	ns, found := namespaces[namespace]
+	return ns, found
+}
+
+func (c *kialiCacheImpl) GetNamespaces(cluster string, token string) ([]models.Namespace, bool) {
+	c.namespacesLock.RLock()
+	defer c.namespacesLock.RUnlock()
+
+	key := namespacesKey{cluster: cluster, token: token}
+	namespaces, found := c.namespaceStore.Get(key)
+
+	return maps.Values(namespaces), found
+}
+
+func (c *kialiCacheImpl) RefreshTokenNamespaces(cluster string) {
+	c.namespacesLock.Lock()
+	defer c.namespacesLock.Unlock()
+
+	for _, key := range c.namespaceStore.Keys() {
+		if key.cluster == cluster {
+			c.namespaceStore.Remove(key)
+		}
+	}
+}
+
+func (c *kialiCacheImpl) SetNamespaces(token string, namespaces []models.Namespace) {
+	c.namespacesLock.Lock()
+	defer c.namespacesLock.Unlock()
+
+	namespacesByCluster := make(map[string][]models.Namespace)
+	for _, namespace := range namespaces {
+		namespacesByCluster[namespace.Cluster] = append(namespacesByCluster[namespace.Cluster], namespace)
+	}
+
+	for cluster, clusterNamespaces := range namespacesByCluster {
+		key := namespacesKey{cluster: cluster, token: token}
```

Comment:
Any need to validate the token here, check for empty, before it gets used in a map key?